src/browser.qml

```diff
@@ -57,4 +93,48 @@ ApplicationWindow {
         property: "mainWindow"
         value: webView
     }
+
+    Component {
+        id: backgroundComponent
+        Browser.Background {
+            id: bg
+            property Item pageContainer
+            property Page page
+
+            // Page and attached background is destroyed when the page gets destroyed.
+            onPageChanged: {
+                if (!page) {
+
+                    var index = 0
+                    for (; index < backgrounds.length; ++index) {
+                        if (backgrounds[index].background === bg) {
+                            break
+                        }
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
+                        backgrounds = backgrounds.slice(index + 1)
+                    } else if (index > 0 && index < backgrounds.length - 1) {
+                        var firstPart = backgrounds.slice(0, index)
+                        var secondPart = backgrounds.slice(index + 1)
+                        backgrounds = firstPart.concat(secondPart)
```

Comment:
Good point... Following should be enough.

```background.splice(index, 1)```

Will verify that original array updates correctly.